wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ViewModel/ConversationListViewControllerViewModel.swift

```diff
@@ -123,7 +123,18 @@ extension ConversationListViewController.ViewModel {
     func setupObservers() {
 
         if let userSession = ZMUserSession.shared() {
-            initialSyncObserverToken = ZMUserSession.addInitialSyncCompletionObserver(self, userSession: userSession)
+            let context = userSession.managedObjectContext
+
+            initialSyncObserverToken = NotificationCenter.default.addObserver(
```

Comment:
From the docs:
You must invoke [removeObserver(_:)](https://developer.apple.com/documentation/foundation/notificationcenter/1413994-removeobserver) or [removeObserver(_:name:object:)](https://developer.apple.com/documentation/foundation/notificationcenter/1407263-removeobserver) before the system deallocates any object that [addObserver(forName:object:queue:using:)](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver) specifies.

In the closure we have a weak `self` reference, however, the `context` is captured.

Also, if we switch accounts, the `notificationContext` is probably deallocated, which violates the API, if I understand correctly ("before the system deallocates any object that addObserver specifies").

Note: if `setupObservers()` is called more than once, the old token is overwritten.